experimental/trusted_shuffler/client/src/main.rs

```diff
@@ -32,27 +35,106 @@ pub struct Opt {
         default_value = "http://localhost:8080"
     )]
     server_url: String,
+    #[structopt(long, help = "The QPS we are aiming to simulate", default_value = "10")]
+    qps: u32,
+    #[structopt(
+        long,
+        help = "How many rounds, i.e., seconds, we sent requests",
+        default_value = "1"
+    )]
+    rounds: u32,
 }
 
-#[tokio::main]
+#[tokio::main(flavor = "multi_thread", worker_threads = 2)]
 async fn main() -> anyhow::Result<()> {
-    env_logger::init();
+    env_logger::builder()
+        .format_timestamp(None)
+        .format_level(false)
+        .format_module_path(false)
+        .format_target(false)
+        .init();
     let opt = Opt::parse();
 
-    let request = "Request 1";
-    info!("Sending request: {}", request);
-    let url = format!("{}/request", &opt.server_url);
-
-    let request_start = Instant::now();
-    let response = send_request(&url, Method::POST, request.as_bytes())
-        .await
-        .context("Couldn't receive response")?;
-    let request_duration = request_start.elapsed();
-
-    let parsed_response = String::from_utf8(response).context("Couldn't decode response body")?;
-    info!(
-        "Received response: {} in {:?}",
-        parsed_response, request_duration
-    );
+    let expected_qps = opt.qps;
+    let rounds = opt.rounds;
+
+    // Start with giving one second iterations.
+    let start_time = Instant::now();
+
+    // Header of table printed by logger.
+    log::info!("phase,id,request_elapsed_in_ms,response_elapsed_in_ms");
```

Comment:
I'm not sure what this means, is the expectation to output something like a CSV? If it's meant to be machine readable, how about outputting it to a separate file as an actual CSV (or perhaps as JSON)? But feel free to follow up in a separate PR (if you think it's useful anyways)

Reply:
This is already a CSV format
It is needed to generate QPS/latency plots

Reply:
But isn't it mixed with the logs? Which means that we need to modify the logger (now I see why), and also not log anything else. I think it would be much easier to emit it to a separate machine readable file, instead of logging.

Reply:
ah, makes sense

Reply:
I would create an mpsc tokio channel somewhere, and let the tasks write onto it, then collect all the values on the other side in a background task, and write them to a file using some CSV library (also, CSV is a surprisingly nontrivial format to implement, I recommend using a library instead of trying to approximate it with format statements)

Reply:
Thank you, we changed the output of the program for now, but we'll come back that, when we want csv output from the program. You are right about csv and logging!